runtime/compiler/il/J9MethodSymbol.cpp

```diff
@@ -121,13 +121,32 @@ J9::MethodSymbol::isPureFunction()
       case TR::java_math_BigInteger_subtract:
       case TR::java_math_BigInteger_multiply:
          */
-      return true;
+         return true;
       default:
          return false;
       }
    return false;
    }
 
+/**
+ * Returns true if the function call will not yield to OSR point.
+ * 
+ * An example of kind of function which can go in the list would be recognized calls with
+ * NOP calls or the one that are guaranteed to be inlined by codegenerator. 
+ */
+bool 
+J9::MethodSymbol::functionCallDoesNotYieldOSR()
+   {
```

Comment:
Given this is an extensible method I think it should delegate up to OMR - if the OMR implementation returns true then return true otherwise then check the J9 specific methods. We may add some non-helpers or similar to this list in due course and those helpers would be added at the OMR layer.

Reply:
I was looking to make the changes, one of the way I was thinking to modify is to check the list in the OpenJ9 layer first (Here) and if call is not in OpenJ9 list, consult OMR. That should serve the purpose as well. I have made changes in https://github.com/eclipse/openj9/pull/8114/commits/c8dd5fc1174104e3417aca96c22b4e91ed4dc13a , Is this something you had in mind?